src/test/java/com/github/tomakehurst/wiremock/RemoveStubMappingAcceptanceTest.java

```diff
@@ -135,6 +135,55 @@ public void removeStubThatDoesNotExists() {
     assertThat(getMatchingStubCount("/stb-1", "/stb-2"), is(2));
   }
 
+  @Test
+  public void removeStubWithUUIDThatExists() {
+
+    UUID id1 = UUID.randomUUID();
+
+    stubFor(get(urlEqualTo("/stub-1")).withId(id1).willReturn(aResponse().withBody("Stub-1-Body")));
+
+    assertThat(testClient.get("/stub-1").content(), is("Stub-1-Body"));
+
+    UUID id2 = UUID.randomUUID();
+    stubFor(get(urlEqualTo("/stub-2")).withId(id2).willReturn(aResponse().withBody("Stub-2-Body")));
+
+    assertThat(testClient.get("/stub-2").content(), is("Stub-2-Body"));
+
+    assertThat(getMatchingStubCount("/stub-1", "/stub-2"), is(2));
+
+    removeStub(id2);
+
+    assertThat(getMatchingStubCount("/stub-1", "/stub-2"), is(1));
+
+    removeStub(id2);
+
+    assertThat(getMatchingStubCount("/stub-1", "/stub-2"), is(0));
+  }
+
+  @Test
+  public void removeStubWithUUIDThatDoesNotExists() {
```

Comment:
Likewise I think this could be much simpler - really just:
1. Add a stub so that there's one in the list
2. Generate a random ID
3. Remove it
4. Assert that no exception was thrown
5. Assert that there's 1 stub in the list